support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -3,174 +3,84 @@ package controllers
-        </script>""")
-      }).withSettingsSurrogateKey
-    } else {
-      Redirect(routes.DigitalSubscriptionController.digitalGeoRedirect(false)).withSettingsSurrogateKey
-    }
-  }
-
-  def digitalGeoRedirect(orderIsAGift: Boolean = false): Action[AnyContent] = geoRedirect(
-    if (orderIsAGift) "subscribe/digital/gift" else "subscribe/digital"
-  )
-
-  def displayForm(orderIsAGift: Boolean): Action[AnyContent] = {
-    implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    if (settings.switches.enableDigitalSubGifting.isOn || !orderIsAGift) {
-      authenticatedAction(subscriptionsClientId).async { implicit request =>
-
-      identityService.getUser(request.user.minimalUser).fold(
-        error => {
-          SafeLogger.error(scrub"Failed to display digital subscriptions form for ${request.user.minimalUser.id} due to error from identityService: $error")
-          Future.successful(InternalServerError)
-        },
-        user => {
-          userHasDigitalSubscription(membersDataService, request.user) map {
-            case true => if (orderIsAGift) Ok(digitalSubscriptionFormHtml(user, orderIsAGift)) else redirectToExistingThankYouPage
-            case _ => Ok(digitalSubscriptionFormHtml(user, orderIsAGift))
-          }
-        }
-      ).flatten.map(_.withSettingsSurrogateKey)
-    }
-    } else {
-      Action(Redirect(routes.DigitalSubscriptionController.digitalGeoRedirect(false)).withSettingsSurrogateKey)
+        </script>"""
+        )
+      }
+      ).withSettingsSurrogateKey
     }
   }
 
-
-  private def digitalSubscriptionFormHtml(idUser: IdUser, orderIsAGift: Boolean)(implicit request: RequestHeader, settings: AllSettings): Html = {
-    val title = if (orderIsAGift) {
-      "Support the Guardian | The Guardian Digital Gift Subscription"
-     } else {
-       "Support the Guardian | The Guardian Digital Subscription"
-     }
-    val id = EmptyDiv("digital-subscription-checkout-page")
-    val js = "digitalSubscriptionCheckoutPage.js"
-    val css = "digitalSubscriptionCheckoutPage.css"
-    val csrf = CSRF.getToken.value
-    val uatMode = testUsers.isTestUser(idUser.publicFields.displayName)
-    val promoCodes = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
-    val v2recaptchaConfigPublicKey = recaptchaConfigProvider.get(uatMode).v2PublicKey
-    val readerType = if (orderIsAGift) Gift else Direct
-
-    subscriptionCheckout(
-      title,
-      id,
-      js,
-      css,
-      fontLoaderBundle,
-      Some(csrf),
-      idUser,
-      uatMode,
-      priceSummaryServiceProvider.forUser(uatMode).getPrices(DigitalPack, promoCodes, readerType),
-      stripeConfigProvider.get(),
-      stripeConfigProvider.get(true),
-      payPalConfigProvider.get(),
-      payPalConfigProvider.get(true),
-      v2recaptchaConfigPublicKey,
-      orderIsAGift
+  private def getPaperHrefLangLinks(orderIsAGift: Boolean): Map[String, String] = {
```

Comment:
I extracted this function to align with weekly, as that's how it was done there and it makes sense.